packages/lib/src/vis/tiles/TiledLayer.tsx

```diff
@@ -1,51 +1,61 @@
-import { sum } from 'lodash';
-import { Suspense } from 'react';
+import { Suspense, useRef } from 'react';
 import { LinearFilter, NearestFilter, Vector2 } from 'three';
+import type { Group } from 'three';
 
 import { useCameraState } from '../hooks';
+import type { Size } from '../models';
 import { useAxisSystemContext } from '../shared/AxisSystemProvider';
 import Tile from './Tile';
 import type { TilesApi } from './api';
 import type { ColorMapProps } from './models';
 import {
   getTileOffsets,
-  getScaledVisibleDomains,
+  getScaledVisibleBox,
   sortTilesByDistanceTo,
 } from './utils';
 
 interface Props extends ColorMapProps {
   api: TilesApi;
   layer: number;
+  size: Size;
 }
 
 function TiledLayer(props: Props) {
-  const { api, layer, ...colorMapProps } = props;
+  const { api, layer, size: meshSize, ...colorMapProps } = props;
 
   const { baseLayerIndex, numLayers, tileSize } = api;
   const layerSize = api.layerSizes[layer];
 
-  const { abscissaConfig, ordinateConfig, visSize } = useAxisSystemContext();
-  const { xVisibleDomain, yVisibleDomain } = useCameraState(
-    (...args) => getScaledVisibleDomains(...args, layerSize),
-    [layerSize]
+  const groupRef = useRef<Group>(null);
+  const { abscissaConfig, ordinateConfig } = useAxisSystemContext();
+  const box = useCameraState(
+    (...args) => getScaledVisibleBox(...args, meshSize, layerSize, groupRef),
+    [meshSize, layerSize, groupRef]
```

Comment:
Should I pass the `ref` here or `ref.current`?

Reply:
The ref (assuming ESLint complains if you don't pass it), as when `ref.current` changes, it does not trigger a re-render anyway.